pipeline/merger/src/main/scala/uk/ac/wellcome/platform/merger/rules/ItemsRule.scala

```diff
@@ -147,4 +155,41 @@ object ItemsRule extends FieldMergeRule with MergerLogging {
         ))
     }
   }
+
+  /**
+    * Sierra records are created from the Sierra / Calm harvest.
+    * We merge the Sierra item ID into the Calm item.
+    *
+    * If an item is digitised via METS, it is linked to the Sierra record.
+    * We merge that into the Calm item.
+    */
+  private val mergeIntoCalmTarget = new PartialRule {
+    val isDefinedForTarget: WorkPredicate = WorkPredicates.calmWork
+    val isDefinedForSource
+      : WorkPredicate = WorkPredicates.metsWork or WorkPredicates.sierraWork
+
+    def rule(target: UnidentifiedWork,
+             sources: NonEmptyList[TransformedBaseWork]): FieldData = {
+
+      // As the Calm transformer always adds 1 item with 1 location, this is safe
+      val calmItem = target.data.items.head
+
+      val metsSources = sources.filter(WorkPredicates.metsWork)
+      val metsDigitalLocations =
+        metsSources.flatMap(_.data.items.flatMap(_.locations.filter(_ match {
```

Comment:
I _think_ that all METS locations are digital locations - do you know if that's the case?

Reply:
Yes.

Reply:
This might've been here from before due to types and/or me wanting to avoid using brain when refactoring the old code

Reply:
The type system hasn't made it very convenient as you need to keep having checks everywhere. We could rewrite the works type to be something like:

```
Work[LocationsType <: List](locations)

CalmWork extends Work[SingleItemList[Location]]
SierraWork extends [List[Locations]]
METSWork extends Work[SingleItemList[Location]]
```

But it's getting very weird at that point.

Reply:
METS as a source is just a digitized item.

This points that the generic work model does not necessarily help at defining our source data.

Reply:
Another way would be send partial information down the pipes from the transformers, but not sure that's a change I want to make.

Reply:
Removed the filter.